ast/stats.go

```diff
@@ -32,13 +32,38 @@ type AnalyzeTableStmt struct {
 	TableNames     []*TableName
 	PartitionNames []model.CIStr
 	IndexNames     []model.CIStr
-	MaxNumBuckets  uint64
+	AnalyzeOpts    []AnalyzeOpt
 
 	// IndexFlag is true when we only analyze indices for a table.
 	IndexFlag   bool
 	Incremental bool
 }
 
+// AnalyzeOptType is the type for analyze options.
+type AnalyzeOptionType int
+
+// Analyze option types.
+const (
+	AnalyzeOptNumBuckets = iota
+	AnalyzeOptNumTopN
+	AnalyzeOptCMSketchDepth
+	AnalyzeOptCMSketchWidth
+)
+
+// AnalyzeOptionString stores the string form of analyze options.
+var AnalyzeOptionString = map[AnalyzeOptionType]string{
```

Comment:
We can use a slice directly?

Reply:
Yes, but `map` could reduce our mental burden by maintaining their relationships explicitly.